vmm/src/vm.rs

```diff
@@ -694,6 +699,15 @@ impl Vm {
             // it will be running with the last configure memory size.
             self.config.lock().unwrap().memory.size = desired_memory;
         }
+
+        if let Some(desired_balloon) = desired_balloon {
+            self.memory_manager
+                .lock()
+                .unwrap()
+                .balloon_resize(desired_balloon << 20)
+                .map_err(Error::MemoryManager)?;
+        }
```

Comment:
You need to store the balloon size to restore it after a reboot. This should be part of a separate commit.
First thing you need is that `balloon_resize()` return the actual size of the balloon that has been calculated based on the `desired_ram_w_balloon`.
Second thing, you need to add a new `balloon_size: Option<u64>` as part of the `MemoryConfig` so that you can update it, and upon a reboot, the VMM should rely on this value to set the balloon.
/cc @rbradford

Reply:
Fixed.